### PR TITLE
[client] Added unit test cases

### PR DESCRIPTION
**Pull request by niknegi**
### Purpose

Linked issue: https://github.com/apache/fluss-rust/issues/309

### Brief change log

Add unit tests for DataType validation edge cases and RowType projection

### Tests

Screenshot 2026-02-13 at 12 08 54 PM

### CODE REVIEW

File: crates/fluss/src/metadata/datatype.rs [hunk 1]
```diff
@@ -1469,3 +1568,79 @@ fn test_timestamp_ltz_invalid_precision() {
...
+    // Note: project_with_field_names filters out non-existent field names using filter_map,
+    // so the result will be an empty row if all field names are non-existent
+    let projected = row_type
+        .project_with_field_names(&["nonexistent".to_string()])
+        .unwrap();
+    assert_eq!(projected.fields().len(), 0);
+
+    // Mixed existing and non-existing: only existing fields are projected
+    let projected = row_type
+        .project_with_field_names(&["id".to_string(), "nonexistent".to_string()])
```
Comment (luoyuxia) [inline crates/fluss/src/metadata/datatype.rs]: I'd argue it should throw exceptioin when the field doesn't exist just like java side. Could you please help fix it?

Comment (niknegi) [reply crates/fluss/src/metadata/datatype.rs]: Yes, checking it!

File: crates/fluss/src/metadata/datatype.rs [hunk 2]
```diff
@@ -1431,6 +1452,65 @@ fn test_decimal_invalid_scale() {
...
+    assert!(result.is_err());
+    assert!(
+        result
+            .unwrap_err()
+            .to_string()
+            .contains("Decimal precision must be between 1 and 38")
+    );
+}
+
+
```
[the remark this answers is not in the document]
Comment (niknegi) [reply crates/fluss/src/metadata/datatype.rs]: Fixed

File: crates/fluss/src/metadata/datatype.rs [hunk 3]
```diff
@@ -1469,3 +1568,79 @@ fn test_timestamp_ltz_invalid_precision() {
...
+    );
+
+    // Valid projection by indices: [0, 2]
+    let projected = row_type.project(&[0, 2]).unwrap();
+    assert_eq!(projected.fields().len(), 2);
+    assert_eq!(projected.fields()[0].name, "id");
+    assert_eq!(projected.fields()[1].name, "age");
+}
+
+#[test]
```
[the remark this answers is not in the document]
Comment (niknegi) [reply crates/fluss/src/metadata/datatype.rs]: Added

File: crates/fluss/src/metadata/datatype.rs [hunk 4]
```diff
@@ -1431,6 +1452,65 @@ fn test_decimal_invalid_scale() {
...
+    // Valid: precision=1, scale=0
+    let result = DecimalType::with_nullable(false, 1, 0);
+    assert!(result.is_ok());
+    let decimal = result.unwrap();
+    assert_eq!(decimal.precision(), 1);
+    assert_eq!(decimal.scale(), 0);
+    assert!(!decimal.is_nullable());
+}
+
+#[test]
```
[the remark this answers is not in the document]
Comment (niknegi) [reply crates/fluss/src/metadata/datatype.rs]: Added

File: crates/fluss/src/metadata/datatype.rs [hunk 5]
```diff
@@ -594,6 +598,10 @@ impl TimeType {
         self.precision
     }
 
+    pub fn is_nullable(&self) -> bool {
+        self.nullable
+    }
+
```
[the remark this answers is not in the document]
Comment (niknegi) [reply crates/fluss/src/metadata/datatype.rs]: Updated the test to use to_string() instead of is_nullable()

File: crates/fluss/src/metadata/datatype.rs (hunk 1 above)
Comment (niknegi) [reply crates/fluss/src/metadata/datatype.rs]: @luoyuxia Added throwing an exception when any field doesn't exist on RowType

**Review by luoyuxia — APPROVED** [review]
@niknegi Thanks for update. LGTM. But ci fails
